cmake/CheckTrezor.cmake

```diff
@@ -0,0 +1,63 @@
+OPTION(USE_DEVICE_TREZOR "Trezor support compilation" ON)
+OPTION(USE_DEVICE_TREZOR_LIBUSB "Trezor LibUSB compilation" ON)
+OPTION(USE_DEVICE_TREZOR_UDP_RELEASE "Trezor UdpTransport in release mode" OFF)
+
+# Use Trezor master switch
+if (USE_DEVICE_TREZOR)
+    # Protobuf is required to build protobuf messages for Trezor
+    include(FindProtobuf OPTIONAL)
+    find_package(Protobuf)
+    if(NOT Protobuf_FOUND)
+        message(STATUS "Could not find Protobuf")
+    endif()
+
+else()
+    message(STATUS "Trezor support disabled by USE_DEVICE_TREZOR")
+endif()
+
+# Try to build protobuf messages
+if(Protobuf_FOUND AND USE_DEVICE_TREZOR)
+    if ("$ENV{PYTHON3}" STREQUAL "")
+        set(PYTHON3 "python3")
```

Comment:
This can fail. On my system for example the default python 3 is `python3.6`. Some people it may simply be `python` symlinked also.

Reply:
In that case Trezor support is disabled and build succeeds anyway. 
As a workaround you can set `PYTHON3` env var.

Reply:
Yes I'm well aware of how to workaround it. Point is I shouldn't have to work around it.

Reply:
@jtgrassie I've updated #4839 with more robust Python check (better cmake). That should make it pretty robust. And still, there is a workaround if all `find_module` fails. This PR is already approved for merge so I don't want to add more changes at this point.

Reply:
> I was literally just about to comment that the python search in #4839 was dependent on cmake 3.12+.

Yep I realized this module is new so I've added also the simple one for cmake 3.0+

> BH though, having the python / protobuf dependency at all is unnecessary. Nicer if the maintainer of Trezor support just runs and commits the generated c++ output, thus removing the need for python or protobuf to build in Trezor support.

I would say this is a matter of personal preferences. We discussed this with @moneromooo-monero and came to the conclusion that generating the messages during build time is quite easy so we can do it like this. 

Protobuf library is required also in runtime so I think if somebody wants to build trezor support it is not that much of a hassle to install protobuf compiler and python, which is almost omnipresent nowadays.